os/fs/driver/mtd/smart.c

```diff
@@ -1686,6 +1692,44 @@ static int smart_set_wear_level(FAR struct smart_struct_s *dev, uint16_t block,
 	return 0;
 }
 #endif
+
+static bool smart_validate_erase_block(FAR struct smart_struct_s *dev)
+{
+	FAR struct smart_sect_header_s header;
+	bool corrupted = false;
+	int i;
+	int ret;
+
+	memcpy(&header, dev->rwbuffer, sizeof(struct smart_sect_header_s));
+
+	/* Check the crc of header first. */
+	ret = smart_validate_crc(dev);
+	if (ret != OK) {
+		if (HEADER_IS_CLEAN(header)) {
+			/* If header is erased state, then check contents */
+			for (i = sizeof(struct smart_sect_header_s); i < dev->sectorsize; i++) {
+				if (dev->rwbuffer[i] != 0xFF) {
```

Comment:
some of board treat 0x00 as a erase state, also there's config for it already, you can use it.

Reply:
You're right. Modified.